xclim/sdba/adjustment.py

```diff
@@ -996,3 +1001,184 @@ def _compute_adjust(ds, dim):
+
+
+        These three steps are repeated a certain number of times, prescribed by argument ``n_iter``. At each
+        iteration, a new random rotation matrix is generated.
+
+        The original algorithm ([Pitie05]_), stops the iteration when some distance score converges. Following
+        [Cannon18_] and the MBCn implementation in [CannonR]_, we instead fix the number of iterations.
+
+        As done by [Cannon18]_, the distance score chosen is the "Energy distance" from [SkezelyRizzo]_
+        (see :py:func:`xclim.sdba.processing.escore`).
+
+        The random matrices are generated following a method laid out by [Mezzadri].
+
+        This is only part of the full MBCn algorithm, see :ref:`The MBCn algorithm` for an example on how
+        to replicate the full method with xclim. This includes a standardization of the simulated data beforehand,
+        an initial univariate adjustment and the reordering of those adjusted series according to the
+        rank structure of the output of this algorithm.
+
+        References
+        ----------
+        .. [Cannon18] Cannon, A. J. (2018). Multivariate quantile mapping bias correction: An N-dimensional probability density function transform for climate model simulations of multiple variables. Climate Dynamics, 50(1), 31–49. https://doi.org/10.1007/s00382-017-3580-6
+        .. [Mezzadri]Mezzadri, F. (2006). How to generate random matrices from the classical compact groups. arXiv preprint math-ph/0609050.
+        .. [Pitie05] Pitie, F., Kokaram, A. C., & Dahyot, R. (2005). N-dimensional probability density function transfer and its application to color transfer. Tenth IEEE International Conference on Computer Vision (ICCV’05) Volume 1, 2, 1434-1439 Vol. 2. https://doi.org/10.1109/ICCV.2005.166
+        .. [SkezelyRizzo] Szekely, G. J. and Rizzo, M. L. (2004) Testing for Equal Distributions in High Dimension, InterStat, November (5)
+        """
+        base_kws or {}
+        if "kind" in base_kws:
+            warn(
+                f'The adjustment kind cannot be controlled when using {self.__class__.__name__}, it defaults to "+".'
+            )
+        base_kws.setdefault("kind", "+")
+
+        super().__init__(
+            base=base,
+            base_kws=base_kws,
+            n_escore=n_escore,
+            n_iter=n_iter,
+        )
+
+    def train_adjust(
+        self,
+        ref: xr.DataArray,
+        hist: xr.DataArray,
+        sim: xr.DataArray,
+        *,
+        pts_dim: str = "variables",
+        adj_kws: Optional[Mapping[str, Any]] = None,
+        rot_matrices: Optional[xr.DataArray] = None,
+    ):
+        # Assuming sim has the same coords as hist
+        # We get the safest new name of the rotated dim.
+        rot_dim = xr.core.utils.get_temp_dimname(
+            set(ref.dims).union(hist.dims).union(sim.dims), pts_dim + "_prime"
+        )
+
+        # Get the rotation matrices
+        rot_matrices = rand_rot_matrix(
```

Comment:
Don't you want to be able to pass an existing rot_matrices object? I'm not sure I see the point except for testing though, so you may just want to remove rot_matrices from the input parameters.

Reply:
Oh woups, you're right, we want this option for testing.